src/posix/platform/radio_spinel.cpp

```diff
@@ -301,6 +311,22 @@ otError RadioSpinel::CheckRadioCapabilities(void)
     return error;
 }
 
+otError RadioSpinel::RestoreDatasetFromNcp(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(!mIsRcp);
+
```

Comment:
Is it possible to call otPlatSettingsInit() and otPlatSettingsDeinit() to directly save dataset in this method? This method can be called by Init().